tests/Browser/FilterPostsTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Tests\Browser;
+
+use Tests\DuskTestCase;
+use Rogue\Models\Signup;
+use Rogue\Models\Post;
+use Laravel\Dusk\Browser;
+use Tests\Browser\Pages\HomePage;
+use Tests\Browser\Pages\CampaignSinglePage;
+// use Tests\Browser\Components\Login;
+
+class FilterPostsTest extends DuskTestCase
+{
+    /**
+     * Test filtering by "Pending" returns pending posts.
+     */
+    public function testFilterPendingPosts()
+    {
+        // Create a signup so that the campaign over page will load.
+        $signup = factory(Signup::class)->create();
+
+        // Create a post so the filter will show results.
+        $signup->posts()->save(factory(Post::class)->make());
+
+        $this->browse(function (Browser $browser) use ($signup) {
+            $browser->visit(new HomePage)
+                    // ->within(new Login, function($browser) {
+                    //     $browser->login($browser);
+                    // })
+                    ->click('@login-button')
+                    ->assertPathIs('/register')
+                    ->clickLink('Log In')
+                    ->type('username', env('NORTHSTAR_EMAIL'))
+                    ->type('password', env('NORTHSTAR_PASSWORD'))
+                    ->press('Log In')
+                    ->assertPathIs('/campaigns')
+                    ->visit('/campaigns/' . $signup->campaign_id)
+                    ->on(new CampaignSinglePage($signup->campaign_id))
+                    ->assertSee('Post Filters')
+                    ->select('select', 'pending')
+                    ->assertSelected('select', 'pending')
+                    ->press('Apply Filters')
+                    ->pause(5000)
+                    // ->screenshot('hello');
+                    ->dump();
```

Comment:
@DFurnes this is still WIP but not sure why this is happening - when I dump out the HTML here, it says that there are no results for this search. However, in line 24, I create a post that has a status of "pending" so this should pop up in the results. Are there any gotchas that I'm not thinking of?

Reply:
I left a comment below, I think the campaign ID wasn't being added to the URL!

Reply:
ah @DFurnes I figured it out! When creating the post, for some reason, the post's `campaign_id` is a string, not an integer, so when it does the query it doesn't find anything. Hmm this is tricky for long term testing bc sometimes the `campaign_id` will be an integer (if coming from P-A) and sometimes it will be a string (if coming from P-N). For now, I think I will cast it to whatever the type the `$signup->campaign_id` is as a workaround.

Reply:
Oooof! That's frustrating! Ideally we want these to always be strings (even if it's a string of numbers), so maybe something we can fix in the model with a [cast](https://laravel.com/docs/5.5/eloquent-mutators#attribute-casting)?

Reply:
oh I love that - thanks @DFurnes !